terraform/03_gke_cluster.tf

```diff
@@ -117,33 +128,62 @@ resource "null_resource" "current_project" {
   }
 }
 
-#resource "null_resource" "sleeping_subprocess" {
-#  provisioner "local-exec" {
-#      command = "sleep 60 >./stdout.log 2>./stderr.log & echo \"sleeping in PID\" $!"
-#  }
-#
-#  depends_on = [google_container_cluster.gke]
-#}
-
-# Setting kubectl context to currently deployed GKE cluster
-resource "null_resource" "set_gke_context" {
-  provisioner "local-exec" {
-    command = "gcloud container clusters get-credentials cloud-ops-sandbox --zone ${element(random_shuffle.zone.result, 0)} --project ${data.google_project.project.project_id}"
-  }
+# Create GSA to allow K8S services to access Google APIs
+resource "google_service_account" "set_gsa" {
+  account_id   = "gke-sa"
+  display_name = "gsa"
+  project = data.google_project.project.project_id
 
   depends_on = [
-    google_container_cluster.gke, 
+    google_container_cluster.gke,
     null_resource.current_project
   ]
 }
 
+# Create GSA/KSA binding: let IAM auth KSAs as a svc.id.goog member name
+resource "google_service_account_iam_binding" "set_gsa_binding" {
+  service_account_id = google_service_account.set_gsa.name
+  role = "roles/iam.workloadIdentityUser"
+
+  members = [
+    "serviceAccount:${data.google_project.project.project_id}.svc.id.goog[default/default]"
+  ]
+
+  depends_on = [google_service_account.set_gsa]
+}
+
+# Annotate KSA
+resource "null_resource" "annotate_ksa" {
+  triggers = {
+    cluster_ep = google_container_cluster.gke.endpoint  #kubernetes cluster endpoint
+  }
+
+  provisioner "local-exec" {
+    command = <<EOT
+      gcloud container clusters get-credentials cloud-ops-sandbox --zone ${element(random_shuffle.zone.result, 0)} --project ${data.google_project.project.project_id}
+      kubectl annotate serviceaccount --namespace default default iam.gke.io/gcp-service-account=${google_service_account.set_gsa.email}
```

Comment:
are there terraform providers available to do this?

Reply:
I checked. There are no terraform resources for just `kubectl annotate`. 

I can use `kubernetes_service_account` if I create a new namespace/KSA, but I'm just using the default k8s namespace for simplicity.